pkgs/build-support/rust/default.nix

```diff
@@ -66,14 +72,126 @@ let
     else ''
       cargoDepsCopy="$sourceRoot/${cargoVendorDir}"
     '';
+  
+  targetIsJSON = stdenv.lib.hasSuffix ".json" target;
+
+  rustSrc = stdenv.mkDerivation {
```

Comment:
We have a rustSrc package in `nixpkgs/pkgs/development/compilers/rust/default.nix` already. Please re-use that.

Reply:
Oh wow, only 17 MB